application/forms/ContactGroupForm.php

```diff
@@ -0,0 +1,300 @@
+                'contactgroup_member',
+                [
+                    'contactgroup_id' => $groupIdentifier,
+                    'contact_id'      => $contactId
+                ]
+            );
+        }
+
+        $this->db->commitTransaction();
+
+        return $groupIdentifier;
+    }
+
+    /**
+     * Edit the contact group
+     *
+     * @return bool False if no changes found, true otherwise
+     */
+    public function editGroup(): bool
+    {
+        $isUpdated = false;
+        $values = $this->getValues();
+
+        $this->db->beginTransaction();
+
+        $storedValues = $this->fetchDbValues();
+
+        if ($values['group_name'] !== $storedValues['group_name']) {
+            $this->db->update(
+                'contactgroup',
+                ['name' => $values['group_name']],
+                ['id = ?' => $this->contactgroupId]
+            );
+
+            $isUpdated = true;
+        }
+
+        $storedContacts =  explode(',', $storedValues['group_members']);
+        $newContacts = explode(',', $values['group_members']);
+
+        $toDelete = array_diff($storedContacts, $newContacts);
+        $toAdd = array_diff($newContacts, $storedContacts);
+
+        if (! empty($toDelete)) {
+            $this->db->delete('contactgroup_member', ['contact_id IN (?)' => $toDelete]);
+
+            $isUpdated = true;
+        }
+
+        if (! empty($toAdd)) {
+            foreach ($toAdd as $contactId) {
+                $this->db->insert(
+                    'contactgroup_member',
+                    [
+                        'contactgroup_id' => $this->contactgroupId,
+                        'contact_id'      => $contactId
+                    ]
+                );
+            }
+
+            $isUpdated = true;
+        }
+
+        $this->db->commitTransaction();
+
+        return $isUpdated;
+    }
+
+    /**
+     * Remove the contact group
+     */
+    public function removeContactgroup(): void
+    {
+        $this->db->beginTransaction();
+
+        $this->db->delete('contactgroup_member', ['contactgroup_id = ?' => $this->contactgroupId]);
+        $this->db->delete('contactgroup', ['id = ?' => $this->contactgroupId]);
+
+        $this->db->commitTransaction();
+    }
+
+    /**
+     * Fetch the values from the database
+     *
+     * @return array
+     */
+    private function fetchDbValues(): array
+    {
+        $query = Contactgroup::on(Database::get())
+            ->columns(['id', 'name'])
+            ->filter(Filter::equal('id', $this->contactgroupId));
+
+        $group = $query->first();
+        if ($group === null) {
+            $this->httpNotFound(t('Contact group not found'));
+        }
+
+        $groupMembers = [];
+        foreach ($group->contact->columns('id') as $contact) {
+            $groupMembers[] = $contact->id;
+        }
+
+        return [
+            'group_name'   => $group->name,
+            'group_members' => implode(',', $groupMembers)
+        ];
+    }
```

Comment:
Find the mistake:

![image](https://github.com/Icinga/icinga-notifications-web/assets/16668527/4dfb87e4-4cfd-4ca1-9790-a2d895683887)